whois-api/src/main/java/net/ripe/db/whois/api/rdap/RedactionObjectMapper.java

```diff
@@ -0,0 +1,101 @@
+        UNSUPPORTED_ENTITIES.put(MNT_ROUTES, new Redaction(new Redaction.Description("Route object creation protection"), new Redaction.Description("No registrant mntner")));
+    }
+
+    private static final Map<AttributeType, Redaction> UNSUPPORTED_VCARDS = Maps.newHashMap();
+    {
+        UNSUPPORTED_VCARDS.put(E_MAIL, new Redaction(new Redaction.Description("e-mail contact information"),new Redaction.Description("Personal data")));
+        UNSUPPORTED_VCARDS.put(NOTIFY, new Redaction(new Redaction.Description("Updates notification e-mail information"), new Redaction.Description("Personal data")));
+    }
+
+    public static String UNSUPPORTED_ENTITIES_SYNTAX = "$.entities[?(@.handle=='%s')]";
+
+    public static String UNSUPPORTED_VCARD_SYNTAX = "$.entities[?(@.roles=='%s')].vcardArray[1][?(@[0]=='%s')]";
+
+    RedactionObjectMapper(final RpslObjectDao rpslObjectDao){
+        this.rpslObjectDao = rpslObjectDao;
+    }
+
+    public List<Redaction> createContactOrEntityRedaction(final List<RpslAttribute> rpslAttributes){
+        // TODO: [MH] Part of this logic can be merge with the logic of contactEntities in RdapObjectMapper. Big refactor to make a clear merge
+        final List<Redaction> redactions = Lists.newArrayList();
+        for (final RpslAttribute rpslAttribute: rpslAttributes) {
+            final AttributeType rpslAttributeType = AttributeType.getByName(rpslAttribute.getKey());
+            if (UNSUPPORTED_ENTITIES.containsKey(rpslAttributeType)){
+                createRedaction(UNSUPPORTED_ENTITIES.get(rpslAttributeType), String.format(UNSUPPORTED_ENTITIES_SYNTAX, rpslAttributeType), redactions);
+            } else if (CONTACT_ATTRIBUTE_TO_ROLE_NAME.containsKey(rpslAttributeType)){
+                checkContactEntityLevelCreation(redactions, rpslAttribute, rpslAttributeType);
+            }
+        }
+        return redactions;
+    }
+
+    private void checkContactEntityLevelCreation(final List<Redaction> redactions, final RpslAttribute rpslAttribute, final AttributeType rpslAttributeType) {
+        final List<RpslObject> referencedRpslObject = getObject(rpslAttributeType.getReferences(), rpslAttribute.getCleanValues());
+        if (!referencedRpslObject.isEmpty()) {
+            for (final Map.Entry<AttributeType, Redaction> unsupportedVcard : UNSUPPORTED_VCARDS.entrySet()) {
+                for (final RpslObject rpslObject : referencedRpslObject) {
+                    if (rpslObject.containsAttribute(unsupportedVcard.getKey())) {
+                        createRedaction(unsupportedVcard.getValue(), String.format(UNSUPPORTED_VCARD_SYNTAX,
+                                        CONTACT_ATTRIBUTE_TO_ROLE_NAME.get(rpslAttributeType).name(), unsupportedVcard.getKey()),
+                                redactions);
+                        break;
```

Comment:
Why are you breaking here? Are you sure there cannot be multiple attribute type in rpslObject? Also same rpslObject can have notify as well as e_mail?

Reply:
I did a refactor here